tests/test_web_app.py

```diff
@@ -164,66 +164,53 @@ def test_watchdog_on(create_app_client):
-            return self.ready
-
-    app.index = FakeIndex()
-    response = client.get("/search_index/ready")
-    assert response.data == b"false"
-
-    app.index.ready = True
-    response = client.get("/search_index/ready")
-    assert response.data == b"true"
-
-
-def test_search_get(create_app_client):
+def test_search_endpoint_ok(create_app_client):
     _, client = create_app_client()
-    response = client.get("/search")
+    with patch.object(web_app.CDPedia, '_search') as mock:
+        mock.return_value = [('testlink', 'testtitle', 'testtext')]
+        response = client.post("/search", data={"keywords": "foo bar"})
     assert response.status_code == 200
+    mock.assert_called_once_with('foo bar')
+    assert b'testlink' in response.data
+    assert b'testtitle' in response.data
+    assert b'testtext' in response.data
 
 
-def test_search_post_url(create_app_client):
+def test_search_endpoint_empty(create_app_client):
     _, client = create_app_client()
-    response = client.post("/search")
+    response = client.post("/search", data={"keywords": ""})
     assert response.status_code == 302
+    assert "http://localhost/" == response.location
 
 
-def test_search_post(create_app_client):
-    _, client = create_app_client()
-    response = client.post("/search", data={"keywords": "a"})
-    assert response.status_code == 302
-    assert "/search/" in response.location
-    response = client.post("/search", data={"keywords": "a"}, follow_redirects=True)
-    assert response.status_code == 200
-
-
-def test_search_term_url(create_app_client):
-    _, client = create_app_client()
-    words = ("foo", "bar")
-    response = client.post("/search", data={"keywords": " ".join(words)})
+def test_search_real_search(create_app_client):
+    app = web_app.create_app(watchdog=None, with_static=False)
 
-    assert response.status_code == 302
-    assert "/search/%s" % "+".join(words) in response.location
-
-    response = client.post(
-        "/search", data={"keywords": " ".join(words)}, follow_redirects=True)
-    assert response.status_code == 200
+    with patch.object(app.index, 'partial_search') as index_mock:
+        index_mock.return_value = [
+            ('t/e/s/testlink1', 'testtitle1', 123, True, 'testtext1'),
+            ('t/e/s/testlink moño', 'testtitle2', 567, False, 'testtext2'),
+        ]
+        result = app._search("foo bar Moño")
+    index_mock.assert_called_once_with(['foo', 'bar', 'mono'])
+    assert result == [
+        ('wiki/testlink1', 'testtitle1', 'testtext1'),
+        ('wiki/testlink%20mo%C3%B1o', 'testtitle2', 'testtext2'),
+    ]
 
 
 def test_search_term_with_slash(create_app_client):
-    _, client = create_app_client()
-    data = {"keywords": "foo/bar"}
-    response = client.post("/search", data=data, follow_redirects=True)
-    assert response.status_code == 200
+    app = web_app.create_app(watchdog=None, with_static=False)
+
+    with patch.object(app.index, 'partial_search') as index_mock:
```

Comment:
there are no use of partial search anymore
`with patch.object(app.index, 'search') as index_mock:`

Reply:
Esto ya está corregido en los commits de abajo.